shell/platform/android/io/flutter/view/AccessibilityBridge.java

```diff
@@ -856,17 +853,32 @@ void updateWith(ByteBuffer buffer, String[] strings) {
 
             final int childCount = buffer.getInt();
             if (childCount == 0) {
-                children = null;
+                childrenInTraversalOrder = null;
+            } else {
+                if (childrenInTraversalOrder == null)
+                    childrenInTraversalOrder = new ArrayList<SemanticsObject>(childCount);
+                else
+                    childrenInTraversalOrder.clear();
+
+                for (int i = 0; i < childCount; ++i) {
+                    SemanticsObject child = getOrCreateObject(buffer.getInt());
+                    child.parent = this;
+                    childrenInTraversalOrder.add(child);
+                }
+            }
+
+            if (childCount == 0) {
+                childrenInHitTestOrder = null;
```

Comment:
nit: I find this slightly easier to process if the `if` branch here would be part of the `if` branch above and same for the `else`.